### DISCUSSION TITLE
Does RESTeasy Client support Apache HttpClient 5?

### DISCUSSION THREAD

**Discussion by sopgreg** [resteasy/resteasy#3049] [Q&A]
We are currently using Apache HttpClient 4.3, but plan to migrate to 5.x

There does not seem to be a `ClientHttpEngine` for version 5, or am I missing something?

Thanks.

**Comment by jamezp** [comment]
Hi @sopgreg,
It currently does not no. It has been discussed, but would require quite a few changes. Is there some specific support you're looking for?

I have played with and considered using the Java 11 `HttpClient` as the default delegate client.

**Reply by ghilainm** [reply]
@jamezp Is there any plan to change the backing http client? The current one has not been updated since 2022.

**Reply by jamezp** [reply]
I apologize. We haven't changed anything by default as we were attempting to be backwards compatible. I've done some work to make it easier to introduce new clients and I've deprecated the current Apache HTTP Client implementation classes.

I'd like to introduce a new client module for the JDK HTTP Client. With a note that it will *not* be able to implement all parts of the Jakarta REST specification for the client implementation.

Then I'd like to move the clients out to their own separate project so users can have the option to chose the client they want and we can version them differently than RESTEasy itself.

**Reply by ghilainm** [reply]
No problem, I fully understand the idea :). I guess that currently there is no workaround for this? 

It is in fact blocking me because of Virtual Thread pinning issue of the apache HTTP client :). > https://issues.apache.org/jira/browse/HTTPCORE-746

**Reply by jamezp** [reply]
You could try using the [Jetty Client](https://github.com/resteasy/resteasy-jetty). All you should need to do is ensure the library is on your class path. I need to write up some better docs on it. There is some limited documentation here https://docs.resteasy.dev/6.2/userguide/#_transport_layer.

**Reply by ghilainm** [reply]
@jamezp Oh nice that really helps, I didn't see the documentation, honestly my bad :). I'll try that tomorrow! Thanks a lot :).

**Reply by jamezp** [reply]
No worries. I didn't publicize it enough nor document it as well I should have :)
